stackslib/src/chainstate/tests/consensus.rs

```diff
@@ -94,993 +90,1292 @@ const fn clarity_versions_for_epoch(epoch: StacksEpochId) -> &'static [ClarityVe
-    default_chain_id: u32,
-}
+        StacksChainState::process_block_transactions(clarity_tx, block_txs, 0)
+            .map_err(|e| e.to_string())?;
 
-impl TestTxFactory {
-    /// Creates a new [`TransactionFactory`] with the specified default chain ID.
-    pub fn new(default_chain_id: u32) -> Self {
-        Self {
-            nonce_counter: HashMap::new(),
-            default_chain_id,
-        }
-    }
+        NakamotoChainState::finish_block(clarity_tx, None, false, burn_header_height)
+            .map_err(|e| e.to_string())?;
 
-    /// Manually increments the nonce for the sender of the specified transaction.
-    ///
-    /// This method should be called *after* a transaction has been successfully
-    /// processed to ensure the factory uses the correct next nonce for subsequent
-    /// transactions from the same sender.
-    ///
-    /// # Arguments
-    ///
-    /// * `tx_spec` - The original specification of the transaction whose sender's
-    ///   nonce should be incremented.
-    ///
-    /// # Panics
-    ///
-    /// Panics if the sender's address is not found in the nonce counter map.
-    pub fn increase_nonce_for_tx(&mut self, tx_spec: &TestTxSpec) {
-        let sender_privk = match tx_spec {
-            TestTxSpec::Transfer { from, .. } => from,
-            TestTxSpec::ContractDeploy { sender, .. } => sender,
-            TestTxSpec::ContractCall { sender, .. } => sender,
-        };
-        let address = StacksAddress::p2pkh(false, &StacksPublicKey::from_private(sender_privk));
-        let nonce = self
-            .nonce_counter
-            .get_mut(&address)
-            .unwrap_or_else(|| panic!("Nonce not found for address {address}"));
-        *nonce += 1;
+        Ok(clarity_tx.seal())
     }
-
-    /// Generates a new transaction of the specified type.
-    ///
-    /// Arguments:
-    /// - `tx_type`: The type of transaction to create.
-    ///
-    /// Returns:
-    /// A [`StacksTransaction`] representing the created transaction.
-    pub fn generate_tx(&mut self, tx_spec: &TestTxSpec) -> StacksTransaction {
-        match tx_spec {
-            TestTxSpec::Transfer { from, to, amount } => self.transfer(from, to, *amount),
-            TestTxSpec::ContractDeploy {
-                sender,
-                name,
-                code,
-                clarity_version,
-            } => self.contract_deploy(sender, name, code, *clarity_version),
-            TestTxSpec::ContractCall {
-                sender,
-                contract_addr,
-                contract_name,
-                function_name,
-                args,
-            } => self.contract_call(sender, contract_addr, contract_name, function_name, args),
+}
+
+/// A complete consensus test that drives a `ConsensusChain` through a series of epochs.
+///
+/// It stores the blocks to execute per epoch and runs them in chronological order,
+/// producing a vector of `ExpectedResult` suitable for snapshot testing.
```

Comment:
nit: for consistency with the rest of the docs
```suggestion
/// A complete consensus test that drives a [`ConsensusChain`] through a series of epochs.
///
/// It stores the blocks to execute per epoch and runs them in chronological order,
/// producing a vector of [`ExpectedResult`] suitable for snapshot testing.
```